script/gcp.sh

```diff
@@ -11,11 +16,48 @@ function deploy_lm-logs {
 	echo "Creating subscription"
 	gcloud pubsub subscriptions create ${TOPIC} --topic=${TOPIC}
 
+	echo "Creating VPC Network"
+	gcloud compute networks create ${VPC_NAME} --subnet-mode=custom
+
+	echo "Creating firewall rule"
+	gcloud compute firewall-rules create ${FIREWALL_RULE} \
+    --network ${VPC_NAME} \
+    --allow tcp:22 \
+    --source-ranges 35.235.240.0/20 \
+    --target-tags allow-ssh
+
+	echo "Creating subnet"
+	gcloud compute networks subnets create ${SUBNET_NAME} \
+    --network=${VPC_NAME} \
+    --range=10.0.0.0/24
+
+    echo "Getting the region"
+    REGION=$(gcloud compute networks subnets list --filter="name=${SUBNET_NAME}" --format="value(region)")
+
+    echo "Creating router"
+    gcloud compute routers create ${ROUTER_NAME} \
+    --network=${VPC_NAME} \
+    --region=${REGION}
+
+    echo "Creating NAT configuration on Router"
+    gcloud compute routers nats create ${NAT_NAME} \
+    --router=${ROUTER_NAME} \
+    --region=${REGION} \
+    --nat-all-subnet-ip-ranges \
+    --auto-allocate-nat-external-ips
+
 	echo "Creating VM"
-	gcloud compute instances create ${NAME} \
-  --image debian-11-bullseye-v20230814 \
-  --image-project debian-cloud \
-	--machine-type=e2-micro
+	gcloud compute instances create ${VM_NAME} \
```

Comment:
will the compute instance be created in the same region where routers, networks are created by default?

Reply:
It will ask region while creating subnet, for VM it will ask about zone which comes under same region :
Creating VM
Did you mean zone [europe-west4-b] for instance: [lm-logs-forwarder] (Y/n)?  Y